### DISCUSSION TITLE
Offline support

### DISCUSSION THREAD

**Discussion by mfulton26** [honojs/hono#1346] [Ideas]
It seems to me that a hono app can serve from the edge a service worker file that bundles the app to run offline in the client's browser so that the app can be used without a network connection. With SQLite WASM and other technologies developers could write code that writes to local databases and then syncs when online via the service worker.

**Comment by yusukebe** [comment]
FYI, Hono works on service workers:

https://github.com/yusukebe/service-worker-magic

**Reply by janat08** [reply]
What does server SW worker mean? Are all routes cached by SW outside of server and sw routes? SW routes are for static pages right? If I update the repo does hono update the version of SW to reset static pages cache?

**Reply by mfulton26** [reply]
This is great @yusukebe. Can this get added to the docs on hono.dev? Or maybe it is there and I'm simply not seeing it yet. Thank you.

**Comment by nickchomey** [comment]
I just made this comment in the [Middleware proposal discussion](https://github.com/honojs/hono/issues/482#issuecomment-2441919002), asking for, essentially, Google Workbox (toolkit for doing PWA stuff) to be replicated in Hono. I think it has tremendous promise and would be happy to help spec and build it if there is interest.
